diff --git a/node_modules/vitepress/dist/client/theme-default/components/VPAlgoliaSearchBox.vue b/node_modules/vitepress/dist/client/theme-default/components/VPAlgoliaSearchBox.vue
index 31bfd64..154189a 100644
--- a/node_modules/vitepress/dist/client/theme-default/components/VPAlgoliaSearchBox.vue
+++ b/node_modules/vitepress/dist/client/theme-default/components/VPAlgoliaSearchBox.vue
@@ -87,6 +87,18 @@ function initialize(userOptions: DefaultTheme.AlgoliaSearchOptions) {
 
 function getRelativePath(url: string) {
   const { pathname, hash } = new URL(url, location.origin)
+
+  const hrefUrl = new URL(window.location.href)
+  if (hrefUrl.hostname.endsWith('github.io')) {
+    const pathSplit = hrefUrl.pathname.split('/')
+    if (pathSplit[1]) {
+      return ('/' + pathSplit[1] + pathname).replace(
+        /\.html$/,
+        site.value.cleanUrls ? '' : '.html'
+      ) + hash
+    }
+  }
+
   return pathname.replace(/\.html$/, site.value.cleanUrls ? '' : '.html') + hash
 }
 </script>
